crates/matrix-sdk-ui/src/timeline/mod.rs

```diff
@@ -568,6 +594,38 @@ impl Timeline {
         SendAttachment::new(self, path.into(), mime_type, config)
     }
 
+    /// Redact an event given its [`TimelineEventItemId`] and an optional
+    /// reason.
+    ///
+    /// See [`Self::redact`] for more info.
+    pub async fn redact_by_id(
+        &self,
+        id: &TimelineEventItemId,
+        reason: Option<&str>,
+    ) -> Result<(), Error> {
+        match id {
+            TimelineEventItemId::TransactionId(transaction_id) => {
+                let Some(event) = self.item_by_transaction_id(transaction_id).await else {
```

Comment:
I think that this reads easier, if we use a `match` and list all the possible branches using the match:

```rust
TimelineEventItemId::TransactionId(transaction_id) => {
    let item = self.item_by_transaction_id(transaction_id).await;
    
    match item.as_ref().map(|i| i.handle()) {
        Some(TimelineItemHandle::Local(handle)) => {
            handle.abort().await.map_err(RoomSendQueueError::StorageError)?;
                Ok(())
            }
            Some(TimelineItemHandle::Remote(_)) => {
                // Log an error here instead of a panic?
                panic!("If the item is local, this should never happen");
            }
            None => Err(Error::RedactError(RedactError::LocalEventNotFound(
                transaction_id.to_owned(),
            ))),
        }
    }
TimelineEventItemId::EventId(event_id) => {
    self.room()
        .redact(event_id, reason, None)
        .await
        .map_err(|e| Error::RedactError(RedactError::HttpError(e)))?;
    
    Ok(())
}
```

Also, that panic scares me a bit, though I understand that it can't currently happen.

Reply:
I can't just log that, I'd have to return an error.

Reply:
Or an `Ok(())`, no?